AGENTS.md

```diff
@@ -0,0 +1,305 @@
+  - `UPSTREAM: <carry>: Skip CPU resource status for workload-pinned pods`
+- **Important**: These create ongoing maintenance burden - only use when the change cannot go upstream
+
+### `UPSTREAM: <drop>:`
+Changes that should NOT be included in future rebases.
+- **When to use**: Any temporary or regeneratable change specific to this branch
+- **Lifecycle**: Dropped during rebases
+- **Common cases**:
+  - Generated files that will be regenerated (`make update`)
+  - Vendored dependencies that will be re-vendored
+  - Behavior changes specific to just this minor release
+  - Branch-specific fixes that won't apply to future versions
+- **Examples**:
+  - `UPSTREAM: <drop>: make update`
+  - `UPSTREAM: <drop>: hack/update-vendor.sh`
+
+### `UPSTREAM: 12345:`
+Cherry-picks from upstream Kubernetes PRs.
+- **When to use**: Backporting an upstream fix before the next rebase
+- **Format**: The number is the upstream PR ID from `kubernetes/kubernetes`
+- **Lifecycle**: Only picked if not yet in the new upstream base
+- **Example**: `UPSTREAM: 134442: Fix ResourceQuota test for CRDs with long names`
+
+### Direct upstream commits (no prefix)
+Commits cherry-picked directly from `kubernetes/kubernetes` with their original commit messages.
+- **When to use**: ONLY during the rebase process itself
+- **AI tools should NEVER suggest these** in regular pull requests
+- These are added by maintainers when rebasing to a new Kubernetes version
+
+### Important Squashing Rules
+- **OpenShift-specific files** (`openshift-hack/`, OpenShift READMEs) should be squashed into a single commit: `UPSTREAM: <carry>: Add OpenShift specific files`
+- **Generated changes** must NEVER be mixed with code changes
+- **Related carries** should be squashed together to simplify future rebases
+
+### Enforcement
+**Pull requests that do not follow these commit message conventions will be rejected by maintainers.** Every commit must use one of the `UPSTREAM:` prefixes listed above (except for direct upstream commits added during rebase).
+
+## Rules for AI Tools Making Changes
+
+### DO NOT Freely Modify Files
+Unlike typical repositories, you cannot simply suggest changes anywhere. You must:
+
+1. **Understand the intent**: Is this fixing an OpenShift-specific issue or an upstream bug?
+2. **Choose the right approach**:
+   - If it's an upstream bug → Should be fixed in `kubernetes/kubernetes` first, then cherry-picked
+   - If it's OpenShift-specific → Use `UPSTREAM: <carry>:` prefix
+   - If it's generated code → Will be handled by `make update` with `UPSTREAM: <drop>:`
+
+3. **Use the correct commit prefix** based on the rules above
+
+### Prefer Upstream Fixes
+If you identify a bug that affects Kubernetes generally (not just OpenShift):
+1. The fix should ideally go to upstream `kubernetes/kubernetes` first
+2. Then cherry-pick to OpenShift using `UPSTREAM: <PR number>:` format
+3. Only use `UPSTREAM: <carry>:` for truly OpenShift-specific behavior
```

Comment:
These are good instrcutions, but I feel like they are not AI-specific. Maybe they would fit better in one of the README's we have (I'm OK with leaving them here as well).